plugins/FrequencyShifter/FrequencyShifterEffect.cpp

```diff
@@ -0,0 +1,294 @@
+	const float glide = m_controls.m_glide.value();
+	const float tone = m_controls.m_tone.value();
+	const float phase = m_controls.m_phase.value() * twoPi;
+
+	const bool resetShifterBtn = m_controls.m_resetShifter.value();
+	const bool resetLfoBtn = m_controls.m_resetLfo.value();
+
+	if (!m_prevResetShifter && resetShifterBtn)
+	{
+		m_phase[0] = 0.f;
+		m_phase[1] = 0.f;
+	}
+	if (!m_prevResetLfo && resetLfoBtn)
+	{
+		m_lfoPhase = 0.f;
+	}
+	m_prevResetShifter = resetShifterBtn;
+	m_prevResetLfo = resetLfoBtn;
+
+	const float twoPiOverSr = twoPi / m_sampleRate;
+	const float invRing = 1.f - ring;
+	const bool parallelFB = (routeMode >= 1);
+	const bool routeAdd = (routeMode == 1);
+
+	const bool doHarm = (harmonics > 0.f);
+	const float harmFactor = harmonics * 20.f + 1.f;
+	const float harmDiv = 1.f / (harmonics * 0.3f + 1.f);
+
+	const float dampCoeff = std::exp(-2.f * std::numbers::pi_v<float> * delayDamp / m_sampleRate);
+	const float toneCoeff = std::exp(-2.f * std::numbers::pi_v<float> * tone / m_sampleRate);
+	const float delayGlideCoeff = delayGlide ? std::exp(-1.f / (delayGlide * m_sampleRate)) : 0.f;
+	const float glideCoeff = glide ? std::exp(-1.f / (glide * m_sampleRate)) : 0.f;
+
+	const int wantSize = static_cast<int>(delayLen) + 4;// +4 provides space for interpolation
+	if (wantSize > m_ringBufSize)
+	{
+		m_ringBufSize = wantSize;
+		m_ringBuf.resize(wantSize);
+	}
+
+	// we only bother with wrapping phases once per buffer
+	m_lfoPhase = std::fmod(m_lfoPhase, twoPi);
+	for (int ch = 0; ch < 2; ++ch)
+	{
+		m_phase[ch] = std::fmod(m_phase[ch], twoPi);
+	}
+
+	for (fpp_t i = 0; i < frames; ++i)
+	{
+		float lfo0;
+		float lfo1;
+		if (lfoAmt > 0.f)
+		{
+			lfo0 = std::sin(m_lfoPhase) * lfoAmt;
+			lfo1 = std::sin(m_lfoPhase + lfoSt) * lfoAmt;
+		}
+		else
+		{
+			lfo0 = 0.f;
+			lfo1 = 0.f;
+		}
+		m_lfoPhase += lfoRate;
+
+		const float base0 = fs - spread;
+		const float base1 = fs + spread;
+		m_trueShift[0] = (1.f - glideCoeff) * base0 + glideCoeff * m_trueShift[0];
+		m_trueShift[1] = (1.f - glideCoeff) * base1 + glideCoeff * m_trueShift[1];
+
+		m_trueDelay = std::max((1.f - delayGlideCoeff) * delayLen + delayGlideCoeff * m_trueDelay, 1.f);
+		m_truePhase = (1.f - glideCoeff) * phase + glideCoeff * m_truePhase;
+
+		float readIndex = static_cast<float>(m_writeIndex) - m_trueDelay;
+		if (readIndex < 0.f) { readIndex += static_cast<float>(m_ringBufSize); }
+
+		const int indexFloor = static_cast<int>(readIndex);
+		const float frac = readIndex - static_cast<float>(indexFloor);
+
+		const std::array<float, 2> dly = getHermiteSample(indexFloor, frac);
+
+		const float inL = buf[i][0];
+		const float inR = buf[i][1];
+		const float fxInL = parallelFB ? (dly[0] * feedback) : (inL + dly[0] * feedback);
+		const float fxInR = parallelFB ? (dly[1] * feedback) : (inR + dly[1] * feedback);
+		
+		++m_writeIndex;
+		if (m_writeIndex == m_ringBufSize)
+		{
+			m_writeIndex = 0;
+		}
```

Comment:
(Non-blocking) There's lots of short single-statement blocks (and functions) that according to our coding conventions can be a single line (and several places where you do actually do this!) Addressing these is entirely optional.

```suggestion
		if (++m_writeIndex == m_ringBufSize) { m_writeIndex = 0; }
```